src/main/kotlin/com/sourcegraph/cody/commands/ui/CommandsTabPanel.kt

```diff
@@ -20,30 +27,39 @@ class CommandsTabPanel(
 
   init {
     layout = BoxLayout(this, BoxLayout.Y_AXIS)
-    CommandId.values().forEach { command -> addButton(command) }
-  }
+    CommandId.values().forEach { command -> addCommandButton(command) }
 
-  private fun executeCommandWithContext(commandId: CommandId) {
-    FileEditorManager.getInstance(project).selectedTextEditor?.let {
-      CodyEditorFactoryListener.Util.informAgentAboutEditorChange(it, hasFileChanged = false) {
-        executeCommand(commandId)
-      }
+    if (ConfigUtil.isFeatureFlagEnabled("cody.feature.inline-edits") ||
+        CodyApplicationSettings.instance.isInlineEditionEnabled) {
+      addInlineEditActionButton("cody.editCodeAction")
+      addInlineEditActionButton("cody.documentCodeAction")
     }
   }
 
-  private fun addButton(commandId: CommandId) {
-    val button = JButton(commandId.displayName)
-    val indexOfFirst = commandId.displayName.indexOfFirst { it == commandId.mnemonic }
-    if (indexOfFirst >= 0) {
-      button.displayedMnemonicIndex = indexOfFirst
-    } else {
-      button.setMnemonic(commandId.mnemonic)
+  private fun addInlineEditActionButton(actionId: String) {
+    val action = ActionManagerEx.getInstanceEx().getAction(actionId)
+    addButton(action.templatePresentation.text, action.templatePresentation.mnemonic) {
+      val editor = FileEditorManager.getInstance(project).selectedTextEditor ?: return@addButton
+      val dataContext = (editor as? EditorEx)?.dataContext ?: return@addButton
+      val managerEx = ActionManagerEx.getInstanceEx()
+      val event =
+          AnActionEvent(null, dataContext, ActionPlaces.UNKNOWN, Presentation(), managerEx, 0)
+      ActionUtil.performActionDumbAwareWithCallbacks(action, event)
     }
+  }
+
+  private fun addCommandButton(commandId: CommandId) {
+    addButton(commandId.displayName, commandId.mnemonic) { executeCommand(commandId) }
+  }
+
+  private fun addButton(displayName: String, mnemonic: Int, action: () -> Unit) {
+    val button = JButton(displayName)
+    button.setMnemonic(mnemonic)
     button.alignmentX = Component.CENTER_ALIGNMENT
     button.maximumSize = Dimension(Int.MAX_VALUE, button.getPreferredSize().height)
     val buttonUI = DarculaButtonUI.createUI(button) as ButtonUI
     button.setUI(buttonUI)
-    button.addActionListener { executeCommandWithContext(commandId) }
+    button.addActionListener { action() }
```

Comment:
can it be `button.addActionListener(action)`?

Reply:
Unfortunately not, `addActionListener` takes `ActionListener l` as parameter and when I'm using `{ }` it does implicit conversion.